angular/src/components/lock.component.ts

```diff
@@ -185,7 +186,11 @@ export class LockComponent implements OnInit {
 
     togglePassword() {
         this.showPassword = !this.showPassword;
-        document.getElementById(this.pinLock ? 'pin' : 'masterPassword').focus();
+        if (this.ngZone.isStable) {
+            document.getElementById(this.pinLock ? 'pin' : 'masterPassword').focus();
```

Comment:
Better to extract this this conditional to its own variable, e.g.
`const input = document.getElementById(this.pinLock ? 'pin' : 'masterPassword');`

To avoid repeating it in the `if` and `else` blocks.